.ci/load/Jenkinsfile

```diff
@@ -0,0 +1,179 @@
+        string(name: "concurrent_requests", defaultValue: "100", description: "The number of concurrent requests to test with")
+        string(name: "duration", defaultValue: "10m", description: "Test duration in minutes. Max: 280 minutes")
+        // num_of_runs currently unsupported
+        // string(name: "num_of_runs", defaultValue: "1", description: "Number of test runs to execute")
+        text(name: "agent_config", "defaultValue": "", description: "Custom APM Agent configuration. (WARNING: May echo to console. Do not supply sensitive data.)")
+        text(name: "locustfile", "defaultValue": "", description: "Locust load-generator plan")
+        booleanParam(name: "local_metrics", description: "Enable local metrics collection?", defaultValue: false)
+        // End script auto-generation
+    }
+
+    stages {
+        stage('Pre-flight'){
+            steps {
+                echo 'Getting authentication information from Vault'
+                withSecretVault(secret: 'secret/apm-team/ci/bandstand', user_var_name: 'APP_TOKEN_TYPE', pass_var_name: 'APP_TOKEN'){
+                    script {
+                        sh(script: ".ci/load/scripts/start.sh", returnStdout: true)
+                        env.LOCUST_RUN_TIME = params.duration
+                        env.LOCUST_USERS = params.concurrent_requests
+                        env.SESSION_TOKEN = sh(script: ".ci/load/scripts/start.sh", returnStdout: true).trim()
+                    }
+                }
+            }    
+        }
+        stage('Load test') {
+            parallel {
+                stage('Load generation') {
+                    agent { label 'metal' }
+                    steps {
+                            withSecretVault(secret: 'secret/apm-team/ci/bandstand', user_var_name: 'APP_TOKEN_TYPE', pass_var_name: 'APP_TOKEN'){
+                                echo 'Preparing load generation..'
+                                script {
+                                    if (params.locustfile) {
+                                        echo 'Using user-supplied plan for load-generation with Locust'
+                                        sh script: "echo \"${params.locustfile}\">.ci/load/scripts/locustfile.py"
+                                    }
+                                }
```

Comment:
```suggestion
                                whenTrue(params.locustfile) {
                                    echo 'Using user-supplied plan for load-generation with Locust'
                                    sh script: "echo \"${params.locustfile}\">.ci/load/scripts/locustfile.py"
                                }
```

Reply:
Ooh, I didn't know about `whenTrue`. That's handy! Thanks!

Reply:
@v1v Setting this actually broke the pipeline. :(

```
Preparing load generation..— Print Message<1s

[2020-11-03T08:22:31.234Z] Preparing load generation..

Scripts not permitted to use method groovy.lang.GroovyObject invokeMethod java.lang.String java.lang.Object (org.jenkinsci.plugins.workflow.cps.CpsClosure2 whenTrue java.lang.String org.jenkinsci.plugins.workflow.cps.CpsClosure2)
```

Reply:
I went ahead and wrapped this in a `Boolean.valueOf()` and things are fine again. 👍

Reply:
Sorry about the error, glad you solved :)